docs/fundamentals/executor/executor-reload.md

```diff
@@ -0,0 +1,94 @@
+(reload-executor)=
+# Reload Executor
+
+While developing your Executor, it can be useful to have the Executor be refreshed from the source code while you are working on it, without needing to restart the complete server.
+
+For this you can use the `reload` argument for the Executor so that it watches changes in the source code and makes sure that the changes are applied live to the served Executor.
+
+The Executor will keep track in changes inside the Executor source file, every file passed in `py_modules` argument from {meth}`~jina.Flow.add` and all the Python files inside the folder where the Executor class is defined and its subfolders.
+
+````{admonition} Note
+:class: note
+This feature is thought to let the developer iterate faster while developing or improving the Executor, but is not intended to be used in production environment.
+````
+
+````{admonition} Note
+:class: note
+This feature requires watchfiles>0.18 package to be installed.
```

Comment:
```suggestion
This feature requires watchfiles>=0.18 package to be installed.
```
and actually I am in favor of using `pip install jina[devel]` for the following reasons:
* we group packages by tags, so users know they don't have to include anything besides jina in their requirements.txt
* the version here will get stale. Better to keep our extra-requirements.txt as the single source of truth

Reply:
I do not like this `devel` thing anymore. This only requires `watchfiles` so adding `devel` seems like a very big overkill